libs/sdk-ui-tests-e2e/cypress/integration/01-sdk-ui-dashboard/dateFiltering.spec.ts

```diff
@@ -0,0 +1,80 @@
+// (C) 2023 GoodData Corporation
+
+import * as Navigation from "../../tools/navigation";
+import { DateFilter } from "../../tools/dateFilter";
+import { EditMode } from "../../tools/editMode";
+import {Kpi} from "../../tools/kpi";
+
+const dateFilter = new DateFilter();
+const editMode = new EditMode();
+
+describe("Date filtering", () => {
+    it("should disable default date correctly", { tags: "checklist_integrated_tiger" }, () => {
+        Navigation.visit("dashboard/new-dashboard");
+        dateFilter.subtitleHasValue("This month");
+    });
+
+    it("should update date filter value correctly", { tags: "checklist_integrated_tiger" }, () => {
+        Navigation.visit("dashboard/dashboard-many-rows-columns");
+        dateFilter.subtitleHasValue("All time");
+        dateFilter.openAndSelectOptionByName("This year").apply().subtitleHasValue("This year");
+    });
+
+    it("should display message on the top date filter dropdown", { tags: "checklist_integrated_tiger" }, () => {
+        Navigation.visit("dashboard/dashboard-many-rows-columns");
+        editMode.edit().isInEditMode(true);
+        dateFilter.open().isTopMessageVisibled(true);
+    });
+
+    it("should reset the unsaved selected date filter when open edit mode", { tags: "checklist_integrated_tiger" }, () => {
+        Navigation.visit("dashboard/dashboard-many-rows-columns");
+        editMode.edit().isInEditMode(true);
+        dateFilter.open().isTopMessageVisibled(true);
+    });
+
+    it("should reset the selected date filter on view mode when open edit mode", { tags: "checklist_integrated_tiger" }, () => {
+        Navigation.visit("dashboard/dashboard-many-rows-columns");
+        dateFilter.openAndSelectOptionByName("This year").apply().subtitleHasValue("This year");
+        editMode.edit().isInEditMode(true);
+        dateFilter.subtitleHasValue("All time");
+    });
+
+    it("should display the selected date interval correctly", { tags: "checklist_integrated_tiger" }, () => {
+        Navigation.visit("dashboard/dashboard-many-rows-columns");
+
+        const dateFilterValues = ["This month", "This year", "All time"];
+
+        dateFilterValues.forEach((dateFilterValue) => {
+            editMode.edit();
+            dateFilter.openAndSelectOptionByName(dateFilterValue).apply();
+            editMode.saveButtonEnabled(true).save(true);
+            dateFilter.open().hasSelectedDateFilterValue(dateFilterValue);
+            Navigation.visit("dashboard/dashboard-many-rows-columns");
+            dateFilter.open().hasSelectedDateFilterValue(dateFilterValue);
+        });
+    });
+
+    it("should select date filter by preset", { tags: "checklist_integrated_bear" }, () => {
+        Navigation.visit("dashboard/for-date-filter");
+
+        const dateFilterValues = [
```

Comment:
export to enum variable then import and re-use again because it is used in many tests

Reply:
updated.